### PR TITLE
WIP: initial implementation of shutdown behavior

### PR DESCRIPTION
**Pull request by fatz**
implements #78 

this is my first shot to implement the shutdown switch. In some internal tests it's looking promising. However i'm currently not sure about the test implementation and open for some suggestions.

### CODE REVIEW

File: runtime/opt/taupage/init.sh [hunk 1]
```diff
@@ -94,5 +94,22 @@ else
...
+if [ "$config_shutdown" ] && [ "$config_shutdown" != "never" ]; then
+  echo "INFO: waiting for taupage"
+  EXITCODE=$(docker wait taupage)
+
+  echo "INFO: taupage ended with status ${EXITCODE}"
+  if [ "$config_shutdown" = "on-exit" ]; then
+    shutdown -h 1 "taupage ended: status ${EXITCODE} calling shutdown in 1"
+  else
+    [ "$EXITCODE" -eq 0 ] && shutdown -h 1 "taupage ended successfully: status ${EXITCODE} calling shutdown in 1"
+    exit §EXITCODE
```
Comment (elezar) [inline runtime/opt/taupage/init.sh]: Typo? `$`?

File: test.sh [hunk 2]
```diff
@@ -187,4 +187,75 @@ for testinstance_type in $testinstance_types; do
 	finally true
 done
 
+# get a batch server
+echo "Starting batch server..."
+result=$(aws ec2 run-instances \
+    --image-id $AMI_ID \
+    --count 1 \
+    --instance-type $instance_type \
+		--instance-initiated-shutdown-behavior terminate\
```
Comment (elezar) [inline test.sh]: whitespace.

File: runtime/opt/taupage/init.sh [hunk 3]
```diff
@@ -94,5 +94,22 @@ else
     echo "ERROR: $RUNTIME failed to start with exit code $result ($ELAPSED_SECONDS seconds elapsed)"
 fi
 
-# finished!
-exit $result
+# FIXME: should be improved by checking for on-exit and on-success
+if [ "$config_shutdown" ] && [ "$config_shutdown" != "never" ]; then
+  echo "INFO: waiting for taupage"
+  EXITCODE=$(docker wait taupage)
```
Comment (hjacobs) [inline runtime/opt/taupage/init.sh]: the Docker container name is "taupageapp"

File: runtime/opt/taupage/init.sh (hunk 1 above)
Comment (fatz) [inline runtime/opt/taupage/init.sh]: for sure thx

File: runtime/opt/taupage/init.sh (hunk 3 above)
Comment (fatz) [inline runtime/opt/taupage/init.sh]: you're right... will fix this. Also found out that it's important to do the test with on-success. Cause `docker wait` exits with 1 of the name is unknown.

**Comment by hjacobs** [comment]
Sorry for the delay, we'll review this PR soon

File: runtime/opt/taupage/init.sh [hunk 4]
```diff
@@ -20,8 +20,8 @@ START_TIME=$(date +"%s")
 cd $(dirname $0)
 
 # general preparation
-for script in $(ls init.d); do
-    ./init.d/$script
+for script in ./init.d/*; do
+    ./$script
```
Comment (szuecs) [inline runtime/opt/taupage/init.sh]: this implementation has a bug if you have for example files with spaces

**Comment by szuecs** [comment]
:+1:

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**